pkg/controller/grouppermission/grouppermission_controller_test.go

```diff
@@ -0,0 +1,344 @@
+		NamespacedName: types.NamespacedName{
+			Name:      "test",
+			Namespace: "",
+		},
+	}
+
+	// process the request
+	res, err := reconciler.Reconcile(request)
+	if res.Requeue {
+		t.Error("Didn't expect to requeue")
+	}
+	if err != nil {
+		t.Errorf("Got an unexpected error: %s", err)
+	}
+}
+
+func TestValidGroupPermission(t *testing.T) {
+	ctx := context.TODO()
+	reconciler := newTestReconciler()
+	defer reset(ctx, reconciler)
+
+	// Register operator types with the runtime scheme.
+	s := scheme.Scheme
+
+	//Add api to scheme
+	if err := routev1.AddToScheme(s); err != nil {
+		t.Fatalf("Unable to add route scheme: (%v)", err)
+	}
+
+	// creates a groupPermission object
+	nerr := reconciler.client.Create(ctx, makeGroupPermission())
+	if nerr != nil {
+		t.Errorf("Couldn't create required GroupPermission object for test: %s", nerr)
+	}
+
+	err := reconciler.client.Create(ctx, mockClusterRole())
+	if err != nil {
+		t.Errorf("Couldn't create clusterRole for test: %s", err)
+	}
+
+	// get empty ClusterRoleList and give it a namespace
+	list := &rbacv1.ClusterRoleList{}
+	opts := realclient.ListOptions{Namespace: ""}
+
+	// create clusterRoleList{}
+	err = reconciler.client.List(ctx, &opts, list)
+	if err != nil {
+		t.Errorf("Couldn't get clusterRoleList for test: %s", err)
+	}
+
+	// create a request - mock the API call
+	request := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      "rbac-permissions-operator",
+			Namespace: "default",
+		},
+	}
+
+	// test to see if populateCrClusterRoleNames returns list of clusterRoleNames
+	// need to pass in GroupPermission and list.. it should return an empty list
+	tmpList := populateCrClusterRoleNames(makeGroupPermission(), list)
+	if tmpList != nil {
+		t.Errorf("this should be an empty list but: %s", tmpList)
+	}
+
+	// process the request
+	res, err := reconciler.Reconcile(request)
+	if res.Requeue {
+		t.Error("Didn't expect to requeue")
+	}
+	if err != nil {
+		t.Errorf("Got an unexpected error: %s", err)
+	}
+}
+
+func TestClusterRoleBinding(t *testing.T) {
+	ctx := context.TODO()
+	reconciler := newTestReconciler()
+	defer reset(ctx, reconciler)
+
+	// Register operator types with the runtime scheme.
+	s := scheme.Scheme
+
+	//Add api to scheme
+	if err := routev1.AddToScheme(s); err != nil {
+		t.Fatalf("Unable to add route scheme: (%v)", err)
+	}
+
+	// creates a groupPermission object
+	nerr := reconciler.client.Create(ctx, makeGroupPermission())
+	if nerr != nil {
+		t.Errorf("Couldn't create required GroupPermission object for test: %s", nerr)
+	}
+
+	// create a request - mock the API call
+	request := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      "rbac-permissions-operator",
+			Namespace: "default",
+		},
+	}
+
+	// make a new clusterRoleBinding by passing in clusterRoleName and groupName
+	// it needs to match our mockClusterRoleName
+	// newClusterRoleBinding := newClusterRoleBinding("exampleClusterRoleName", "exampleGroupName")
```

Comment:
Should this be uncommented?

Reply:
This is the test I have not been able to figure out. When logging the newClusterRoleBinding and mockClusterRoleBinding, everything seems to match but I'm still getting an error. Have not been to able to figure out why.

Reply:
was comparing memory address instead of actual content. fixed.